chore: fix indentation and JSDoc comments that eslint is warning about

bradenmacdonald opened the pull request:
## Description

I'm seeing warnings like this on every PR to edx-platform on GitHub:

![Screenshot showing check failures in unchanged files](https://github.com/user-attachments/assets/b4672341-da3f-421b-afa1-3e3854aee2cc)

I thought the fix is trivial. I addressed all the eslint warnings, but then when I pushed the PR changes, more warnings showed up. And so on. It turns out there are over 600 eslint warnings in the codebase right now, though they're not blocking any PRs. For now, I'm just fixing what I can in this PR.

## Supporting information

The correct formatting for JSDoc comments can be seen at e.g. https://jsdoc.app/about-getting-started

## Testing instructions

n/a, no code changes

## Deadline

None

bradenmacdonald committed:
chore: fix indentation and JSDoc comments that eslint is warning about

openedx-webhooks commented:
Thanks for the pull request, @bradenmacdonald!


This repository is currently maintained by `@openedx/wg-maintenance-edx-platform`.


Once you've gone through the following steps feel free to tag them in a comment and let them know that your changes are ready for engineering review.

<details><summary>:radio_button: Get product approval</summary>

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.
</details>

<details><summary>:radio_button: Provide context</summary>

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads
</details>





<details><summary>:radio_button: Get a green build</summary>

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.
</details><details>


---

<details><summary>Where can I find more information?</summary>

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)
</details>

<details><summary>When can I expect my changes to be merged?</summary>

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*
</details>
<!-- comment:external_pr -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

bradenmacdonald committed:
chore: fix more eslint warnings

bradenmacdonald committed:
chore: fix lint warnings in course_outline.js

robrap commented:
I appreciate the clean up. Some questions, some of which I'll also take elsewhere and get back to you:
1. How do we want to handle large formatting changes? Related, are we certain we want all of these rules?
2. Do we have any rule around total error count that can be reduced with this change so that these don't all just reappear?
3. Do want want to amnesty linting issues so we can make a zero tolerance rule?

robrap reviewed (approved):
1. Approving these changes.
2. I plan to remove the tests and add an ADR.
Thank you.

bradenmacdonald committed:
chore: decrease eslint violation limit

bradenmacdonald commented:
@robrap Thanks for reviewing!

> How do we want to handle large formatting changes?

IMHO the best way is to run an automated formatter, like prettier, configured to match our rules, then commit that. In this case, I think we can actually run `eslint --fix` and then commit that, although that's not what I did. It would make a larger PR but should more or less solve the problem. When I started working on this PR, I didn't realize how many fixes were required.

> Related, are we certain we want all of these rules?

The fixes in this PR are mostly indentation, and the indentation is definitely wrong. While some style rules are very debatable, I think consistent indentation is important so I definitely support that rule at least.

> Do we have any rule around total error count that can be reduced with this change so that these don't all just reappear?

Ah, I didn't realize that but we do: https://github.com/openedx/edx-platform/blob/a112a53e75375686291dbc8e139bc3d96577df58/scripts/eslint.py#L28

I rebased this and added b80d0e02710a44df5353d2caaa18821870af3b1c to lower the limit.

> Do want want to amnesty linting issues so we can make a zero tolerance rule?

That would be better, but I think automated lint fixing would be even better still. Certainly the indentation issues can all be fixed automatically.

robrap commented:
Thanks @bradenmacdonald.

After discussing with @feanil and @arbrandes, we thought that we might want an ADR to remove ESLint from edx-platform entirely because:
1. Partly, there was an assumption that it was broken. But it seems the threshold is just higher than the number of violations.
2. Mainly, because all frontend code in edx-platform is theoretically deprecated (although I'm not sure if there is a DEPR or ADR for this?). If this is true, it seems like your time (and other's) might be better spent on higher priority issues.
3. The fact that github is bringing these issues up on PRs is again raising priority on something we don't necessarily want people to be distracted by.

We had also discussed ignoring individual ESLint rules, one of which was indentation, and it seems you might disagree with that particular thought. This is all moot if we disable entirely, which may be the only way to resolve point 3 above.

Given all this, what are your thoughts @bradenmacdonald? Thanks.

bradenmacdonald commented:
@robrap I'd still like to merge this PR since it's ready and it does clean up some code, but yeah - I agree that since we're trying to get rid of all the JS code in edx-platform and the lint rules aren't really affecting whether or not the existing code _works_, it makes a lot of sense to just remove eslint from edx-platform. I'd love to eventually see us get rid of any need for Node or JS in edx-platform entirely. 👍🏻
